drivers/media/platform/bcm2835/bcm2835-unicam.c

```diff
@@ -2137,13 +2137,13 @@ static int unicam_open(struct file *file)
 		goto unlock;
 	}
 
-	if (!v4l2_fh_is_singular_file(file))
-		goto unlock;
-
-	ret = v4l2_subdev_call(dev->sensor, core, s_power, 1);
-	if (ret < 0 && ret != -ENOIOCTLCMD) {
-		v4l2_fh_release(file);
-		goto unlock;
+	if (v4l2_fh_is_singular_file(file)) {
+		/* Only the first open instance must power the subdev. */
+		ret = v4l2_subdev_call(dev->sensor, core, s_power, 1);
+		if (ret < 0 && ret != -ENOIOCTLCMD) {
+			v4l2_fh_release(file);
+			goto unlock;
+		}
```

Comment:
I actually disagree with doing it this way. 
The v4l2_fh_open call is the one that makes the node open or not. s_power is then a secondary operation. Just move the node->open++ to before the v4l2_fh_is_singular_file check.

Reply:
And I'm now wondering if there is a better way to use the fh_list within struct video_device to determine if there are open handles or not in unicam_all_nodes_streaming, rather than adding our own reference counting. One for later.

Reply:
But if we fail to power the subdev, do we leave the node open?

PS, should we abandon this PR and use #3583?

Reply:
It needs the node->open--; that we removed under review ;-)

#3583 is merged (imx219 g_selection). I think you meant #3585, and yes.